### ISSUE TITLE
Handle status tracing history differently

### ISSUE THREAD

**Issue by ir-fuel** [corona-warn-app/cwa-wishlist#154]
## Avoid duplicates

* [X] This issue has not already been raised before
* [X] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation

Currently the status tracing history is being stored in the sqlite store, which also means that that history is deleted when someone deletes the app.

The problem is that deleting and reinstalling the app removes that history, even if the exposure keys were generated.

e.g. install the app, run it 2 weeks. Delete the app and reinstall the app. It will show "unknown risk" since logging hasn't been turned on long enough.

## Suggested Enhancement

Calling the EN framework to get the diagnosis keys, check their dates and fill out the status tracing history on first startup.

## Expected Benefits

No longer need to wait before the app becomes usable. (And I presume that if another app was used before those keys will already be there too).

Disclaimer: I didn't check if deleting the app deletes the diagnosis keys internal to the device.

**Comment by GPclips** [comment]
This is related to https://github.com/corona-warn-app/cwa-wishlist/issues/8

**Comment by ir-fuel** [comment]
I don't think so, and moreover it's specific to the iOS implementation. The #8 feature talks about taking backups and moving to a new phone. What I describe is on the same phone.

**Comment by GPclips** [comment]
Hello @ir-fuel,

yes, of course, it is different but has some similar touchpoints. I just linked it for our development team so that they have all possibly related issues available. I personally like your idea and I am about to transfer it as an enhancement to our development team. You just replied so fast, that had no chance to transfer it. :-)


Best Regards,
LMM

Corona-Warn-App Open Source Team

**Comment by thomasaugsten** [comment]
Hi ir-fuel thanks for your input but the issue with recovering the history is we don't have access to the diagnoses keys saved on a secure partition of the phone.
We have only the date of the downloaded keys but the date of the diagnosis keys of the user is here important. We can only access the diagnosis key if the user give the explicit consent to share the keys with the app and we only allow to ask the user when he wants to submit the keys because of a positive test result.

**Comment by ir-fuel** [comment]
Ah yes indeed. And asking this at first startup could become confusing for the user.

Hmmm, then you would have to write something in a place that is not deleted when uninstalling an app.

**Comment by thomasaugsten** [comment]
From data privacy perspective the user has the right and we the responsibility to delete all information of the app if the user resets or uninstall the app

**Comment by larswmh** [comment]
Closing this issue as the related internal ticket has been set to obsolete
